ComfortZoneApp/app/src/main/java/com/example/comfortzone/Utils/LocationUtil.java

```diff
@@ -0,0 +1,96 @@
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+
+public class LocationUtil {
+
+    public static final String TAG = "locationutil";
+    public static final int PERMISSION_ID = 44;
+    public static String lat;
+    public static String lon;
+
+
+    @SuppressLint("MissingPermission")
+    public static void getLastLocation(Activity activity, Context context, FusedLocationProviderClient fusedLocationClient) {
+        if (checkPermissions(context)) {
+            if (isLocationEnabled(context)) {
+                fusedLocationClient.getLastLocation().addOnCompleteListener(new OnCompleteListener<Location>() {
+                    @Override
+                    public void onComplete(@NonNull Task<Location> task) {
+                        Location location = task.getResult();
+                        if (location == null) {
+                            requestNewLocationData(context, fusedLocationClient);
+                        } else {
+                            lat = String.valueOf(location.getLatitude());
+                            lon = String.valueOf(location.getLongitude());
+                        }
+                    }
+                });
+            } else {
+                Toast.makeText(context, "please turn on your location", Toast.LENGTH_LONG).show();
+                Intent intent = new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS);
+                context.startActivity(intent);
+            }
+        } else {
+            requestPermissions(activity);
+        }
+    }
+
+    private static boolean checkPermissions(Context context) {
+        return ActivityCompat.checkSelfPermission(context,
+                Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_GRANTED;
+    }
+
+    private static void requestPermissions(Activity activity) {
+        ActivityCompat.requestPermissions(activity, new String[]{
+                Manifest.permission.ACCESS_COARSE_LOCATION}, PERMISSION_ID);
+    }
+
+    private static boolean isLocationEnabled(Context context) {
+        LocationManager locationManager = (LocationManager) context.getSystemService(context.LOCATION_SERVICE);
+        return locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER) || locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER);
+    }
+
+    @SuppressLint("MissingPermission")
+    private static void requestNewLocationData(Context context, FusedLocationProviderClient fusedLocationClient) {
+        LocationRequest locationRequest = new LocationRequest();
+        locationRequest.setPriority(LocationRequest.PRIORITY_LOW_POWER);
+        locationRequest.setInterval(5);
+        locationRequest.setFastestInterval(0);
+        locationRequest.setNumUpdates(1);
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(context);
+        fusedLocationClient.requestLocationUpdates(locationRequest, locationCallback, Looper.myLooper());
```

Comment:
periodically requesting location will drain the battery quick, and this is every 5 milliseconds :o I think you can use `getCurrentLocation` for single location update: [link](https://developers.google.com/android/reference/com/google/android/gms/location/FusedLocationProviderClient#getCurrentLocation(int,%20com.google.android.gms.tasks.CancellationToken))